serving/src/main/java/ai/djl/serving/http/KServeRequestHandler.java

```diff
@@ -0,0 +1,155 @@
+        }
+        return false;
+    }
+
+    @Override
+    protected void handleRequest(
+            ChannelHandlerContext ctx,
+            FullHttpRequest req,
+            QueryStringDecoder decoder,
+            String[] segments)
+            throws ModelException {
+        HttpMethod method = req.method();
+
+        if (HttpMethod.GET.equals(method) && isKServeDescribeModelReq(segments, method)) {
+            handleKServeDescribeModel(ctx, segments);
+        }
+    }
+
+    private boolean isKServeDescribeModelReq(String[] segments, HttpMethod method) {
+        return HttpMethod.GET.equals(method) && segments.length == 4
+                || (segments.length == 6 && "version".equals(segments[4]))
+                        && "v2".equals(segments[1])
+                        && "models".equals(segments[2]);
+    }
+
+    private void handleKServeDescribeModel(ChannelHandlerContext ctx, String[] segments) {
+        String modelName = segments[3];
+        String modelVersion = null;
+        if (segments.length > 4) {
+            modelVersion = segments[5];
+        }
+
+        ModelManager modelManager = ModelManager.getInstance();
+        Map<String, Endpoint> endpoints = modelManager.getEndpoints();
+
+        Endpoint endpoint = endpoints.get(modelName);
+        if (endpoint == null) {
+            sendModelNotFoundError(modelName, modelVersion, ctx);
+            return;
+        }
+
+        List<Workflow> workflows = endpoint.getWorkflows();
+        if (workflows.isEmpty()) {
+            sendModelNotFoundError(modelName, modelVersion, ctx);
+            return;
+        }
+
+        List<ModelInfo<Input, Output>> models =
+                workflows.stream()
+                        .flatMap(w -> w.getModels().stream())
+                        .collect(Collectors.toList());
+
+        List<String> versions =
+                models.stream()
+                        .map(ModelInfo::getVersion)
+                        .filter(Objects::nonNull)
+                        .collect(Collectors.toList());
+
+        ModelInfo<Input, Output> modelInfo = models.get(0);
+
+        KServeDescribeModelResponse response = new KServeDescribeModelResponse();
+        response.setVersions(versions);
+        response.setPlatformForEngineName(modelInfo.getEngineName());
+
+        ZooModel<Input, Output> model = modelInfo.getModel(modelInfo.withDefaultDevice(null));
+        response.setName(model.getName());
+        DataType dataType = model.getDataType();
+
+        if (model.describeInput() == null || model.describeOutput() == null) {
+            String errorMsg =
+                    "Input/Output shapes are unknown, "
+                            + "please run predict or forward once and call describe model again";
+
+            Map<String, String> content = new ConcurrentHashMap<>();
+            content.put("error", errorMsg);
+
+            NettyUtils.sendJsonResponse(ctx, content, HttpResponseStatus.NOT_FOUND);
```

Comment:
What error code should we return if Input/Ouput is required? Not all the model has that information in DJL?

Reply:
They did not seem to specify any specific error code. 

I saw ONNX returns an empty object when input/output is not available. So I thought null check means, then initialization through prediction is required. 

If this is not supported by all models, then Maybe we can return  input/output, if it is available, and return empty otherwise.